RickLeinecker/Fortuna
# Updated logic for updating inventory on tank edit.

**Description**
Tank edit API also updates user inventory.
* Tank is queried and saved to a variable to get its prior state
* User is queried and saved to a variable to update its inventory.
* User inventory is updated based on tank prior components and future components.
* User is saved.
* Tank is updated with properties from the body.
* Tank is saved.

Resolves #165 

**Type of change**
Check options that are relevant.


**Steps to Verify Functionality**
1. With a user with a populated inventory, update a tank that's been either created or exists and has items on it using the tankUpdate route on Postman.
2. Observe changes in user's inventory based on what's used and removed.

**Final Checklist:**
Go down the list and check them off.

Passes Flow;
![image](https://user-images.githubusercontent.com/24559442/77596621-e73ddc80-6ed2-11ea-941b-e156c5fdbd90.png)
